tools/gcs-bucket-mover/gcs_bucket_mover/bucket_mover_service.py

```diff
@@ -0,0 +1,657 @@
+            source_bucket_name, sink_bucket_name, target_project),
+        'status':
+        'ENABLED',
+        'projectId':
+        target_project,
+        'schedule': {
+            'scheduleStartDate': {
+                'day': now.day - 1,
+                'month': now.month,
+                'year': now.year
+            },
+            'scheduleEndDate': {
+                'day': now.day - 1,
+                'month': now.month,
+                'year': now.year
+            }
+        },
+        'transferSpec': {
+            'gcsDataSource': {
+                'bucketName': source_bucket_name
+            },
+            'gcsDataSink': {
+                'bucketName': sink_bucket_name
+            },
+            "transferOptions": {
+                "deleteObjectsFromSourceAfterTransfer": True,
+            }
+        }
+    }
+    result = sts_client.transferJobs().create(body=transfer_job).execute(
+        num_retries=5)
+    return result['name']
+
+
+def _check_sts_job(spinner, sts_client, target_project, job_name):
+    """Check on the status of the STS job.
+
+    Args:
+        spinner: The spinner displayed in the console
+        sts_client: The STS client object to be used
+        target_project: The name of the target project where the STS job will be created
+        job_name: The name of the STS job that was created
+
+    Returns:
+        The status of the job as an StsJobStatus enum
+    """
+
+    filter_string = (
+        '{{"project_id": "{project_id}", "job_names": ["{job_name}"]}}').format(
+            project_id=target_project, job_name=job_name)
+
+    result = sts_client.transferOperations().list(
+        name='transferOperations', filter=filter_string).execute(num_retries=5)
+
+    if result:
+        operation = result['operations'][0]
+        metadata = operation['metadata']
+        if 'done' in operation:
+            if operation['done']:
+                if metadata['status'] != 'SUCCESS':
+                    spinner.fail('X')
+                    return sts_job_status.StsJobStatus.failed
+
+                _print_sts_counters(spinner, metadata['counters'], True)
+                spinner.ok(CHECKMARK)
+                return sts_job_status.StsJobStatus.success
+        else:
+            # Update the status of the copy
+            if 'counters' in metadata:
+                _print_sts_counters(spinner, metadata['counters'], False)
+
+    return sts_job_status.StsJobStatus.in_progress
+
+
+def _print_sts_counters(spinner, counters, is_job_done):
+    """Print out the current STS job counters.
+
+    Args:
+        spinner: The spinner displayed in the console
+        counters: The counters object returned as part of the STS job status query
+        is_job_done: If True, print out the final counters instead of just the in progress ones
+    """
+
+    if counters:
+        bytes_copied_to_sink = '0'
+        if 'bytesCopiedToSink' in counters:
```

Comment:
```
bytes_copied_to_sink = counters.get('bytesCopiedToSink', 0)
```

That pattern will simplify this whole block a lot.

Reply:
This is a MUCH nicer pattern.